### PR TITLE
Feat/notifications tests

### PR DESCRIPTION
**Pull request by alexanderleegs**
## Problem

This PR introduces tests into the base notifications PR, #508 

This PR also modifies our test command to add the `--runInBand` flag to prevent jest from running integration tests in parallel (see [link](https://stackoverflow.com/questions/47834131/sequelize-sync-force-true-sometimes-fails-to-set-up-database-correctly)), and also cleans up tables using `sync` between tests to prevent duplicate entries from being created.

### CODE REVIEW

File: src/routes/v2/authenticatedSites/__tests__/Notifications.spec.ts [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+      expect(resp.body).toStrictEqual(mockNotificationsValue)
+      expect(mockNotificationsService.listRecent).toHaveBeenCalledWith({
+        siteName: mockSiteName,
+        userId: mockIsomerUserId,
+      })
+    })
+  })
+
+  describe("getAllNotifications", () => {
+    it("should retrieve the recent notifications for a user and site", async () => {
```
Comment (seaerchin) [inline src/routes/v2/authenticatedSites/__tests__/Notifications.spec.ts:68]: see above comment wrt changing description to reflect that we are calling the underlyign

File: src/routes/v2/authenticatedSites/__tests__/Notifications.spec.ts [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+      expect(resp.body).toStrictEqual(mockNotificationsValue)
+      expect(mockNotificationsService.listAll).toHaveBeenCalledWith({
+        siteName: mockSiteName,
+        userId: mockIsomerUserId,
+      })
+    })
+  })
+
+  describe("markNotificationsAsRead", () => {
+    it("should mark notifications as read", async () => {
```
Comment (seaerchin) [inline src/routes/v2/authenticatedSites/__tests__/Notifications.spec.ts:90]: see above

File: src/fixtures/notifications.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
+const createdTime = new Date()
+const readTime = new Date()
+const oldReadTime = new Date("1995-12-17T03:24:00")
```
[the remark this answers is not in the document]
Comment (alexanderleegs) [reply src/fixtures/notifications.ts]: https://github.com/isomerpages/isomercms-backend/pull/514/commits/a9577c18005a0ff330937286746b08e7cc6aba76

File: src/integration/Notifications.spec.ts [hunk 4]
```diff
@@ -0,0 +1,433 @@
...
+  next()
+})
+subrouter.use(notificationsSubrouter)
+const app = generateRouter(subrouter)
+
+describe("Notifications Router", () => {
+  const mockAdditionalUserId = "2"
+  const mockAdditionalSiteId = "2"
+  const mockAdditionalSiteMemberId = "2"
+  const mockAnotherSiteMemberId = "3"
```
[the remark this answers is not in the document]
Comment (alexanderleegs) [reply src/integration/Notifications.spec.ts]: https://github.com/isomerpages/isomercms-backend/pull/514/commits/a9577c18005a0ff330937286746b08e7cc6aba76

File: src/integration/Notifications.spec.ts [hunk 5]
```diff
@@ -0,0 +1,433 @@
...
+      })
+      await Notification.create({
+        userId: mockIsomerUserId,
+        siteId: mockAdditionalSiteId,
+        siteMemberId: mockAnotherSiteMemberId,
+        ...normalPriorityUnreadNotification,
+      })
+      const expected = [
+        highPriorityUnreadNotification,
+        highPriorityUnreadNotification,
```
[the remark this answers is not in the document]
Comment (alexanderleegs) [reply src/integration/Notifications.spec.ts:272]: Kind of, unread notifs always go first, then priority > creation time for unread, creation time > priority for read

File: src/integration/Notifications.spec.ts [hunk 6]
```diff
@@ -0,0 +1,433 @@
...
+      const actual = await request(app).post("/").send({})
+
+      // Assert
+      expect(actual.statusCode).toBe(expected)
+      expect(
+        await Notification.findAll({
+          where: { siteMemberId: 1, first_read_time: null },
+        })
+      ).toEqual([])
+    })
```
[the remark this answers is not in the document]
Comment (alexanderleegs) [reply src/integration/Notifications.spec.ts:436]: Whoops this is a test description error! The test is supposed to change all unread notifications to read, so the end result is that it shouldn't be able to find any notifications of the target user with no `first_read_time`

File: src/integration/Notifications.spec.ts [hunk 7]
```diff
@@ -0,0 +1,433 @@
...
+        siteMemberId: mockAdditionalSiteMemberId,
+        ...normalPriorityUnreadNotification,
+      })
+      await Notification.create({
+        userId: mockIsomerUserId,
+        siteId: mockAdditionalSiteId,
+        siteMemberId: mockAnotherSiteMemberId,
+        ...normalPriorityUnreadNotification,
+      })
+      const expected = [
```
[the remark this answers is not in the document]
Comment (alexanderleegs) [reply src/integration/Notifications.spec.ts:350]: Makes sense, https://github.com/isomerpages/isomercms-backend/pull/514/commits/bb3cb2e4eb0dd389731cc596133dabb4eef8b063

File: src/routes/v2/authenticatedSites/__tests__/Notifications.spec.ts [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+  )
+
+  const app = generateRouter(subrouter)
+
+  beforeEach(() => {
+    jest.clearAllMocks()
+  })
+
+  describe("getRecentNotifications", () => {
+    it("should retrieve the recent notifications for a user and site", async () => {
```
[the remark this answers is not in the document]
Comment (alexanderleegs) [reply src/routes/v2/authenticatedSites/__tests__/Notifications.spec.ts]: Sure, https://github.com/isomerpages/isomercms-backend/pull/514/commits/bb3cb2e4eb0dd389731cc596133dabb4eef8b063

File: src/integration/Notifications.spec.ts [hunk 9]
```diff
@@ -0,0 +1,433 @@
+import express from "express"
```
[the remark this answers is not in the document]
Comment (alexanderleegs) [reply src/integration/Notifications.spec.ts:1]: Hmm we're not using `ResponseErrorBody` for this router yet, so I opted to leave it out in this PR

File: src/routes/v2/authenticatedSites/__tests__/Notifications.spec.ts (hunk 1 above)
Comment (alexanderleegs) [reply src/routes/v2/authenticatedSites/__tests__/Notifications.spec.ts]: https://github.com/isomerpages/isomercms-backend/pull/514/commits/bb3cb2e4eb0dd389731cc596133dabb4eef8b063

File: src/routes/v2/authenticatedSites/__tests__/Notifications.spec.ts (hunk 2 above)
Comment (alexanderleegs) [reply src/routes/v2/authenticatedSites/__tests__/Notifications.spec.ts]: https://github.com/isomerpages/isomercms-backend/pull/514/commits/bb3cb2e4eb0dd389731cc596133dabb4eef8b063

**Review by seaerchin — APPROVED** [review]
lgtm
